plugins/block-shareable-procedures/src/observable_parameter_model.ts

```diff
@@ -35,13 +35,15 @@ export class ObservableParameterModel implements
   /**
    * Sets the name of this parameter to the given name.
    * @param name The string to set the name to.
+   * @param id The optional ID the backing variable should have.
    * @returns This parameter model.
    */
-  setName(name: string): this {
+  setName(name: string, id?: string): this {
```

Comment:
Is this problem specific to the `ObservableParameterModel`, or put another way why not add this to the interface? Is it because the interface has already been released in core? If it is, I'm willing to use the "new APIs are beta" rule because there is 0 chance anyone has implemented these interfaces.

Reply:
This is specific to the `ObservableParameterModel`. `ObservableParameterModel`s are backed by global variables, while `IParameterModel`s do not assume this (because using global variables to implement parameters is kiiinda sketchy). Adding an optional ID parameter to `setName` on the `IParameterModel` in core would add an assumption of using backing variable models to the interface in core.